### ISSUE TITLE
Visually indicate when mail server connection is broken

### ISSUE THREAD

**Issue by oskarth** [status-im/status-react#3424]
### User Story

As a user, I want to know the reason for messages not arriving.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

*Summary*: Any individual mail server is currently not High Availability, which means that if it goes down messages aren't received. We want to have some kind of visual indicator, such as a shaded topbar or a green/red light in mail server settings that indicate that connection has been broken. A simple ping to the mailserver should suffice.

Similarly, indicate when no peers are connected (`admin.peers` or so).

#### Expected behavior
Get visual feedback if mail server is down.

#### Actual behavior
No visual feedback.

**Comment by oskarth** [comment]
TIME LIMITED BOUNTY UNTIL MAR 3 (just before our offsite). Includes core contributors.  Marking as XL to reflect priority.

It is reasonable to assume this has been the cause of at least some of the "messages are unreliable" problems. It is time-critical for offsite and should be fairly straightforward fix.

cc @andmironov if you can whip up super quick design for how you'd prefer this feedback to be received. (But this issue is *not* blocked by design)

cc @adambabik @JekaMas @dmitryn @b00ris if you want to add something to above.

**Comment by adambabik** [comment]
I think this may be a similar case to a situation when there are no peers connected and this can happen when e.g. system clock is skewed. Up to you if it should be a separate issue or included in this one.

**Comment by status-open-bounty** [comment]
Balance: 0.000000 ETH
Tokens: SNT: 6000.00
Contract address: [0x8693a11ad744196b6421798d7391375b57f31296](https://etherscan.io/address/0x8693a11ad744196b6421798d7391375b57f31296)
Network: Mainnet
Paid to: yenda
Visit [https://openbounty.status.im](https://openbounty.status.im) to learn more.

**Comment by Future-Tomorrow** [comment]
Before diving into the user journey can you tell me where you started from? I don't see a traditional hamburger menu or "messages" as a label and bucket item in the bottom nav. I have some concerns that this doesn't align with a users current mental model of various types of navigation systems and "messages" in the bottom tray would more clearly align with system design heuristics and reachable design principles.

**Comment by oskarth** [comment]
@adambabik Added.

@Future-Tomorrow Sorry I'm not sure I follow. This is about giving user some form of feedback on messages not arriving, so it'd be either anywhere in app as a top bar or so, or possibly as feedback in mailserver settings (visible in nightly build under profile settings IIRC).

**Comment by Future-Tomorrow** [comment]
@oskarth Hi Oscar. Does Status have a html styleguide so someone responding to the bounty can see what was done before for the sake of consistency? This type of message generally belongs in a system of alerts and notifications so curious what those look like?

**Comment by oskarth** [comment]
@Future-Tomorrow No style guide right now, but we have some similar messages for when you are offline, or that you are on testnet not mainnet, etc. But I think anything normal and non-obtrusive would be useful here.

cc @andmironov (lead designer)

**Comment by denis-sharypin** [comment]
Hey @oskarth,
Can we use something like this as a quick fix here? We use this bar when we are showing that app os offline. I added 'refresh' icon and tap on it (the whole row should be tappable) should ping mail server and reconnect to it.
![connection problem](https://user-images.githubusercontent.com/6559881/36682704-1e598c26-1b1c-11e8-8d62-a18c94055338.png)

**Comment by oskarth** [comment]
@denis-sharypin Seems good, not sure about the 'refresh' icon - there's no general simple way of just "reconnecting" unfortunately.

**Comment by denis-sharypin** [comment]
@oskarth can a user do something to fix this issue from her side?

**Comment by denis-sharypin** [comment]
@oskarth  Let's use for this case `Messages connection problem` without any extra icons.  Then we will revisit all errors here https://github.com/status-im/status-react/issues/3015

**Comment by yenda** [comment]
There are 3 differents cases here:
- connecting to mailserver (up to 100 attemps), means it might work in a moment
- mailserver disconnected, I am not sure what this one will mean because we only request messages after login so if mailserver disconnects after sending us the messages we missed it doesn't really matter.
- no peers connected, you won't even be able to send messages so quite similar to No network

I find `Messages connection problem` very vague, at this point maybe we should just say `Error: Can't (send and) receive messages, please restart the app` ?

**Comment by pedropombeiro** [comment]
Reopening since the fix is temporary

**Comment by oskarth** [comment]
@PombeirP @yenda  In what way is the fix temporary? If it is just:

>  The long term solution should be to be able to register a callback on status-go for mailserver disconnection and 0 peers so that we don't have to poll for it.

I think the solution in PR above is Good Enough. Are there any other major problems you see?

**Comment by pedropombeiro** [comment]
@oskarth Yeah, I got that impression from the PR description, so wanted to make sure we didn't overlook the issue in the future. If we're happy with how things are, then it's fine to close this by me.

**Comment by oskarth** [comment]
Makes sense. I think it is good enough for now, might want to tweak messaging. @yenda should get his reward for pushing this through weekend before offsite!

cc @andytudhope

**Comment by yenda** [comment]
@oskarth The current solution is polling `admin_peers` through JSON-RPC which is not ideal and results in a delay up to 30 secs for the message to appear/disappear. I created a new issue on status-go side asking for the implementation of signals or callbacks for these events https://github.com/status-im/status-go/issues/728
